### PR TITLE
feat: refresh withdrawals ui to add clarity around fees (OTE-276)

### PR DESCRIPTION
**Pull request by yogurtandjam**
Following pictures are new UI according to [this design spec](https://www.notion.so/dydx/Revamp-withdrawal-and-deposits-UI-71d417f9c08d4c8eab2de4b27f88cbc5?pvs=4#1cb26156b9a645ba9562fece48d52203)

NOTE: there are 2 minor differences:
1. we don't insert an "and" between the 2nd to last and last cctp token names when listing them in the withdraw form subheader tooltip. This is solely because it's slightly annoying add an 'and' there. I can update this if people feel strongly. Personally, I don't think it matters.
2. We use the term bridge fee instead of 3rd party fees because there's a precedent for a string key const existing in our state. I couldn't find an equivalent for 3rd party fee so I'm opting to keep it as is. I don't think this is an easily fixable change (but I could be wrong).

Screenshot 2024-04-25 at 1 18 03 PM
Screenshot 2024-04-25 at 1 18 20 PM
Screenshot 2024-04-25 at 1 18 25 PM

NOTE: ethereum should have a "lowest fees" slot after but the ethereum chain in our test data doesn't have the correct token address (it uses "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE" instead)
Screenshot 2024-04-25 at 2 03 20 PM

Clarify lowest fee path for withdrawals
https://linear.app/dydx/issue/OTE-276/clarify-lowest-fee-path-for-withdrawals
---

## Views

* `TokenSelectMenu`
  * Adds lowest fees indicator in slotAfter param for the searchSelect dropdown if token is cctp token.

* `WithdrawForm`
  * Adds subheader with tooltip to highlight lowest fee withdrawal path.
  * updates receipt area to always show all available data, reorganize receipt data according to design spec

## Constants/Types

* `constants/cctp`
  * create new file for cctp token constants
  * hopefully we can abstract away the config file from components and views completely and use a consts/utils file

### CODE REVIEW

File: src/views/forms/AccountManagementForms/TokenSelectMenu.tsx [hunk 1]
```diff
@@ -24,6 +25,11 @@ type ElementProps = {
   isExchange?: boolean;
 };
 
+export const tokenIsCCTPToken = (token: SelectionOption): boolean => {
```
Comment (yogurtandjam) [inline src/views/forms/AccountManagementForms/TokenSelectMenu.tsx:28]: [discussion] anyone have any opinions on creating a `utils/cctp` file and moving this there?

**Comment by yogurtandjam** [comment]
[discussion] i'd like to write a small test to make sure the function `tokenIsCCTPToken` doesn't break and works under various edge cases but we don't have a precedent for unit testing. what are people's thoughts on introducing unit testing?

File: src/views/forms/AccountManagementForms/WithdrawForm/WithdrawButtonAndReceipt.tsx [hunk 2]
```diff
@@ -63,38 +60,35 @@ export const WithdrawButtonAndReceipt = ({
...
-      key: 'total-fees',
-      label: <span>{stringGetter({ key: STRING_KEYS.TOTAL_FEES })}</span>,
-      value: <Output type={OutputType.Fiat} value={totalFees} />,
-      subitems: feeSubitems,
+      key: 'expected-amount-received',
+      label: (
+        <span>
+          {stringGetter({ key: STRING_KEYS.EXPECTED_AMOUNT_RECEIVED })}{' '}
+          {withdrawToken && <Tag>{withdrawToken?.symbol}</Tag>}
+        </span>
```
Comment (aforaleka) [inline src/views/forms/AccountManagementForms/WithdrawForm/WithdrawButtonAndReceipt.tsx:70]: instead of adding `{' '}`, can we style the `span` with `layoutMixins.row` so there's a gap instead?

Comment (yogurtandjam) [reply src/views/forms/AccountManagementForms/WithdrawForm/WithdrawButtonAndReceipt.tsx:70]: 100% - for single style items do you mind using inline styles? or do we prefer to always create a styled component?

Comment (yogurtandjam) [reply src/views/forms/AccountManagementForms/WithdrawForm/WithdrawButtonAndReceipt.tsx:70]: nevermind, i'm going to make a component. i see this same issue pops up in `minimum amount received`

File: src/constants/cctp.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import cctpTokens from '../../public/configs/cctp.json';
+
+export const CCTP_TOKEN_NAMES_LOWER_CASE = cctpTokens.map((token) => token.name.toLowerCase());
```
Comment (rosepuppy) [inline src/constants/cctp.ts:3]: CCTP tokens are actually identified via the address not the name in the json.

The name actually refers to the chain name.

File: src/views/forms/AccountManagementForms/TokenSelectMenu.tsx [hunk 4]
```diff
@@ -24,6 +25,11 @@ type ElementProps = {
...
+  if (!token?.stringKey) return false;
+  return CCTP_TOKEN_NAMES_LOWER_CASE.includes(token.stringKey.toLowerCase());
```
Comment (rosepuppy) [inline src/views/forms/AccountManagementForms/TokenSelectMenu.tsx:30]: update to use token.value

File: src/views/forms/AccountManagementForms/TokenSelectMenu.tsx [hunk 5]
```diff
@@ -32,15 +38,13 @@ export const TokenSelectMenu = ({ selectedToken, onSelectToken, isExchange }: El
 
   const tokens =
     (type === TransferType.deposit ? depositOptions : withdrawalOptions)?.assets?.toArray() || [];
-
   const cctpTokensByAddress = cctpTokens.reduce((acc, token) => {
```
Comment (rosepuppy) [inline src/views/forms/AccountManagementForms/TokenSelectMenu.tsx:41]: we can get rid of this now that you made `tokenIsCCTPToken`.

**Review by rosepuppy — COMMENTED** [review]
Awesome work so far! 

Added a few comments, and also needs to do the localization.

This can be a follow up, the design looks like we want to highlight the chain drop down as well:
Screenshot 2024-04-24 at 3 26 32 PM

File: src/views/forms/AccountManagementForms/TokenSelectMenu.tsx (hunk 4 above)
Comment (yogurtandjam) [reply src/views/forms/AccountManagementForms/TokenSelectMenu.tsx]: decided to get rid of this in favor of the existing address object

File: src/constants/cctp.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import cctpTokens from '../../public/configs/cctp.json';
+
+export const CCTP_CHAIN_NAMES_LOWER_CASE = cctpTokens.map((token) => token.name.toLowerCase());
+
+// TODO: make a general capitalize util fn
+export const CCTP_CHAIN_NAMES_CAPITALIZED = CCTP_CHAIN_NAMES_LOWER_CASE.map(
```
[the remark this answers is not in the document]
Comment (yogurtandjam) [reply src/constants/cctp.ts]: oh great comment. i actually just realized this and was going to ask you but you're way ahead of me. thanks man!

**Review by rosepuppy — APPROVED** [review]
LGTM! thanks for iterating!
